fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/join/JoinOrder.java

```diff
@@ -299,31 +299,20 @@ protected void computeCost(ExpressionInfo exprInfo) {
     protected Optional<ExpressionInfo> buildJoinExpr(GroupInfo leftGroup, GroupInfo rightGroup) {
         ExpressionInfo leftExprInfo = leftGroup.bestExprInfo;
         ExpressionInfo rightExprInfo = rightGroup.bestExprInfo;
-        Pair<ScalarOperator, ScalarOperator> predicates = buildInnerJoinPredicate(leftGroup.atoms, rightGroup.atoms);
-
-        ScalarOperator eqPredicate = predicates.first;
-        ScalarOperator otherPredicate = predicates.second;
+        ScalarOperator onPredicates = buildInnerJoinPredicate(leftGroup.atoms, rightGroup.atoms);
```

Comment:
maybe no eqPredicate?

Reply:
Good point. I move this to the computeCost step. When no eqPredicate but with other predicate we add a nestloop join punishment.